sysutils/fontmatrix/Portfile

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           active_variants 1.1
+PortGroup           cmake 1.1
+PortGroup           github 1.0
+
+if {${os.platform} eq "darwin" && ${os.major} < 11} {
+    # Until this is fixed: https://github.com/fontmatrix/fontmatrix/issues/40
+    PortGroup       qt4 1.0
+
+    github.setup    fontmatrix fontmatrix 55f688aafb7c07899fade13dd8d8ec8663127e35
+    version         0.9.99
+    revision        0
+    checksums       rmd160  9f4b52ebce86ca35bd1e442c773c79b758e0e388 \
+                    sha256  5c10a0e466de71c85f652005d2ba1a5b3b898c47b37528d9f7364fcc2539d2a7 \
+                    size    2110252
+} else {
+    PortGroup       qt5 1.0
+
+    github.setup    fontmatrix fontmatrix 082f9379ad33987619addd790f473030da6deabe
+    version         0.9.100
+    revision        0
+    checksums       rmd160  3e817fb8416f3c595065036f5ce61dc5ba46a045 \
+                    sha256  4bac46279e36b633e2c91dd59e874cb6ca352871195df0a6e62949354087331c \
+                    size    2113615
+
+    qt5.depends_component qtlocation qttools qttranslations qtwebkit
+
+    pre-fetch {
+        # https://trac.macports.org/ticket/68508
+        if {[catch {set result [active_variants qt5-qtlocation ""]}]} {
+            if {![catch {set result [active_variants rapidjson-devel ""]}]} {
+                ui_msg "Error: ${name} requires qt5-qtlocation, which does not build with rapidjson-devel.\
+                    Please deactivate rapidjson-devel and install rapidjson."
+                return -code error "Unsupported dependency version."
+            }
+        }
+    }
```

Comment:
this `pre-fetch` doesn't look like the correct solution. If `qt5-qtlocation` doesn't build with `rapidjson-devel` then that dependency in the `qt5-qtlocation` port should be fixed instead of these "hacky" solutions in other ports.

Reply:
@reneeotten Well, there is a ticket for that. The maintainer hasn’t addressed it.

This “hack” here merely ensures the needed version of rapidjson is used, since otherwise the cause of the failure is not at all obvious. It does not modify any code.

Reply:
not sure how `rapidjson-devel` would come into the mix.... as far as I can tell it's not even a dependency of `qt5-qtlocation`.

Reply:
@reneeotten See how it fails: https://trac.macports.org/ticket/68508
I deactivated -devel one, activated regular `rapidjson`, and the failure is gone.

I did not analyze how specifically that works (and do not plan to, tbh), but as a matter of fact, it seems to be the issue.

Reply:
I will drop the check and add a comment note instead.